content/blog/llm-agentic-tool-mesh-orchestrating-agentic-tools-for-the-next-revolution.md

```diff
@@ -0,0 +1,193 @@
+```
+
+## Tool Server Service
+
+The Tool Server Service provides the necessary infrastructure to manage and run LLM Agentic Tool Mesh tools on the platform. It includes capabilities for tool discovery and execution, ensuring that tools are easily accessible and efficiently managed.
+
+Key features:
+
+* Tool discovery: Automatically discover tools within the platform.
+* Execution management: Manage the execution of tools, ensuring efficient operation.
+
+![](/img/tools.png)
+
+Example usage:
+
+```python
+from athon.agents import ToolRepository
+from athon.system import ToolDiscovery
+
+projects_config = [
+    {
+        "name": "Simple Project",
+        "tools": [
+            "examples/local_tool",          # A local tool path
+            "https://127.0.0.1:5003/",      # A remote tool URL
+        ]
+    }
+]
+tools_config = {
+    "type": "LangChainStructured"
+}
+
+def discover_and_load_tools(projects_config, tools_config):
+    tool_repository = ToolRepository.create(tools_config)
+    tool_discovery = ToolDiscovery()
+    tool_id_counter = 1
+
+    for project in projects_config:
+        for tool_reference in project["tools"]:
+            tool_info = tool_discovery.discover_tool(tool_reference)
+            if tool_info:
+                tool_metadata = {
+                    "id": tool_id_counter,
+                    "project": project["name"],
+                    "name": tool_info["name"],
+                    "interface": tool_info.get("interface")
+                }
+                tool_repository.add_tool(tool_info["tool"], tool_metadata)
+                tool_id_counter += 1
+
+    return tool_repository
+
+# Run the tool discovery and loading process
+tool_repository = discover_and_load_tools(projects_config, tools_config)
+
+# Display the discovered tools
+for tool in tool_repository.get_tools().tools:
+    print(f"Discovered tool: {tool['name']} from project: {tool['metadata']['project']}")
+```
+
+# Building a mesh of LLM Agentic Tools
+
+We have developed a series of web applications and tools, complete with examples, to demonstrate the capabilities of LLM Agentic Tool Mesh in out [GitHub repo](https://github.com/HewlettPackard/llmesh).
+
+Web Applications:
+
+* **Chatbot** (`examples/app_chatbot`): A chatbot capable of reasoning and invoking appropriate LLM tools to perform specific actions. You can configure the chatbot using files that define LLM Agentic Tool Mesh platform services, project settings, toolkits, and memory configurations. The web app orchestrates both local and remote LLM tools, allowing them to define their own HTML interfaces, supporting text, images, and code presentations.
+* **Admin panel** (`examples/app_backpanel`): An admin panel that enables the configuration of basic LLM tools to perform actions via LLM calls. It allows you to set the system prompt, select the LLM model, and define the LLM tool interface, simplifying the process of configuring LLM tool interfaces.
```

Comment:
Similarly here, you could say:
Admin panel: There is an admin panel that enables the configuration...

or

Admin panel: An admin panel enables the configuration...